EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+     *
+     * @param whitelistedAddr The address to set as default.
+     * @param rate The personalised rate for donation.
+     */
+    function setSpecificRate(address whitelistedAddr , uint256 rate) external;
+
+    /**
+    *@notice Set for a user a default charity address that will receive donation. 
+    * The default rate specified in {whitelistedRate} will be applied.
+    * @dev Requirements:
+     *
+     * - `whitelistedAddr` cannot be the zero address.
+     *
+     * @param whitelistedAddr The address to set as default.
+     */
+    function setSpecificDefaultAddress(address whitelistedAddr) external;
+
+    /**
+    *@notice Set for a user a default charity address that will receive donation. 
+    * The rate is specified by the user.
+    * @dev Requirements:
+     *
+     * - `whitelistedAddr` cannot be the zero address.
+     * - `rate` cannot be inferior to the default rate 
+     * or to the rate specified by the owner of this contract in {whitelistedRate}.
+     *
+     * @param whitelistedAddr The address to set as default.
+     * @param rate The personalised rate for donation.
+     */
+    function setSpecificDefaultAddressAndRate(address whitelistedAddr , uint256 rate) external;
+
+    /**
+    *@notice Display for a user the default charity address that will receive donation. 
+    * The default rate specified in {whitelistedRate} will be applied.
+     */
+    function specificDefaultAddress() external view returns (
+        address defaultAddress
+    );
+
+    /**
+    *@notice Delete The Default Address and so deactivate donnations .
+     */
+    function deleteDefaultAddress() external;
+}
+
+interface IERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+
+```
+
+To create and deploy an `ERC20Charity` contract one MAY only inherit directly from this [EIP-20](./eip-20.md) extension that would directly signal support for `ERC20Charity` :
+
+```solidity
+pragma solidity ^0.8.4;
+import "./ERC20Charity.sol";
+
+contract CharityToken is ERC20Charity {
+    // ...
+    constructor() ERC20("TestToken", "TST") { }
+    // ...
+}
+```
+
+checking if a contract implement this specification can be done with :
+
+```solidity
+//import "./IERC165.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+
+bytes4 private constant _INTERFACE_ID_ERCcharity = type(IERC20charity).interfaceId; // 0x557512b6
+
+function checkInterface(address testContract) external returns (bool) {
+    (bool success) = IERC165(testContract).supportsInterface(_INTERFACE_ID_ERCcharity);
+    return success;
+}
+```
```

Comment:
```suggestion
```

An EIP doesn't need to walk developers through implementing an interface. You should stick to defining how EIP-6353 tokens behave, for the most part.